kinetics_MO.py

```diff
@@ -0,0 +1,261 @@
+import numpy as np
+import pandas as pd
+import pyomo.environ as pyo
+from measure_optimize import (
+    MeasurementOptimizer,
+    SensitivityData,
+    MeasurementData,
+    CovarianceStructure,
+    ObjectiveLib,
+)
+import pickle
+import time
+
+
+# set up problem formulation
+
+# number of time points for DCM
+Nt = 8
+
+# maximum manual measurement number for each measurement
+max_manual_num = 10
+# minimal measurement interval
+min_interval_num = 10.0
+# maximum manual measurement number for all measurements
+total_max_manual_num = 10
+
+# index of columns of SCM and DCM in Q
+static_ind = [0, 1, 2]
+dynamic_ind = [3, 4, 5]
+# this index is the number of SCM + nubmer of DCM, not number of DCM timepoints
+all_ind = static_ind + dynamic_ind
+num_total_measure = len(all_ind)
+
+# meausrement names
+all_names_strategy3 = [
+    "CA.static",
+    "CB.static",
+    "CC.static",
+    "CA.dynamic",
+    "CB.dynamic",
+    "CC.dynamic",
+]
+
+# define static costs
+static_cost = [2000, 2000, 2000, 200, 200, 200]  # CA  # CB  # CC  # CA  # CB  # CC
+
+# each static-cost measure has no per-sample cost
+dynamic_cost = [0] * len(static_ind)
+# each dynamic-cost measure costs $ 400 per sample
+dynamic_cost.extend([400] * len(dynamic_ind))
+
+# error covariance matrix
+error_cov = [
+    [1, 0.1, 0.1, 1, 0.1, 0.1],
+    [0.1, 4, 0.5, 0.1, 4, 0.5],
+    [0.1, 0.5, 8, 0.1, 0.5, 8],
+    [1, 0.1, 0.1, 1, 0.1, 0.1],
+    [0.1, 4, 0.5, 0.1, 4, 0.5],
+    [0.1, 0.5, 8, 0.1, 0.5, 8],
+]
+
+## change the correlation of DCM VS SCM to half the original value
+# variance
+var_list = [1, 4, 8, 1, 4, 8]
```

Comment:
Why not instead extract this from the diagonal of error_cov?

Reply:
done